### DISCUSSION TITLE
warn - You have enabled the JIT engine which is currently in preview.

### DISCUSSION THREAD

**Discussion by plakhin** [tailwindlabs/tailwindcss#5005] [Help] [answered]
I'm getting this warning when compiling assets. Is this normal behavior since JIT is now a part of the core as of v2.2?
```
warn - You have enabled the JIT engine which is currently in preview.
warn - Preview features are not covered by semver, may introduce breaking changes, and can change at any time.
```

**Comment by olets** [comment, marked as answer]
It's still in preview. From the docs (https://tailwindcss.com/docs/just-in-time-mode):

> This feature is currently in preview. Preview features are not covered by semantic versioning and some details may change as we continue to refine them.

![screenshot of the warning on the JIT docs page](https://user-images.githubusercontent.com/3282350/126543957-2daf2735-f0a4-4b1b-9904-5630e84362c7.png)

**Reply by simonswiss** [reply]
Correct: JIT mode has been added to core in 2.2, but is still technically in preview. The nice thing about it is, if you do find an issue somewhere, you'll discover it at dev time, immediately. 

In a sense, it feels safer than having everything work at dev time, and some issues (missing classes) happening when building for production and purging, in the non JIT mode.

Since you need to define the files to watch _right away_ to use JIT, it usually exposes problems upfront. In that way, it's pretty safe to use today, even if it's in preview! But as the docs say, changes may occur if we uncover good reasons to do so, and that's why the feature is still in preview.

Hope it makes sense!

**Comment by plakhin** [comment]
This definitely makes sense. Just thought it is already out of preview. Thanks for your answers!
